Project/Source/Resources/ResourceClip.cpp

```diff
@@ -120,7 +120,11 @@ void ResourceClip::SetBeginIndex(unsigned int index) {
 	if (endIndex >= index) {
 		beginIndex = index;
 		keyFramesSize = endIndex - beginIndex;
```

Comment:
keyFramesSize is a confusing name. As you see in line 126, you use both keyFramesSize and aniamtion->keyFrames.size(). I'd rename it as something like clipSize (or even clipKeyFramesSize, if you want to maintain "keyFramesSize" in the name)